### ISSUE TITLE
[BUG] Crash on Guake terminal

### ISSUE THREAD

**Issue by Phil995511** [aristocratos/bashtop#141]
**Describe the bug**

Crash continually on Guake Terminal v.3.4.0 (http://guake-project.org) & LXTerminal 0.3.2 on Debian 10 Buster Cinnamon environment and Ubuntu 20.04 Cinnamon environment.

**To Reproduce**

Run Guake Terminal after in Guake run Bashtop 

**Expected behavior**

Not crash's

**Screenshots**

Not applicable

**Info (please complete the following information):**

- Debian and Ubuntu source : 

```
add-apt-repository "deb http://ppa.launchpad.net/bashtop-monitor/bashtop/ubuntu bionic main"
#add-apt-repository "deb-src http://ppa.launchpad.net/bashtop-monitor/bashtop/ubuntu bionic main"

apt-key adv --keyserver keyserver.ubuntu.com --recv-keys 88415F60C906D17BDA2D5E8E6231E014A295AF2A

apt update && apt install bashtop
```

 - Bashtop version : 0.9.11+git229-f48b59e~ubuntu18.04.1 (display v. 0.9.12 in menu options ?!?)

 - (Linux) Linux distribution and version: Debian 10 & Ubuntu 20.04

 - Terminal used: Guake terminal v 3.4.0 & LXTerminal 0.3.2

 - Bash version, "bash --version" (version 4.4 or above is required): GNU bash, version 5.0.3(1)-release (x86_64-pc-linux-gnu)

 - Locales: output of "locale -v" :

LANG=fr_CH.UTF-8
LANGUAGE=fr_CH:fr
LC_CTYPE="fr_CH.UTF-8"
LC_NUMERIC="fr_CH.UTF-8"
LC_TIME="fr_CH.UTF-8"
LC_COLLATE="fr_CH.UTF-8"
LC_MONETARY="fr_CH.UTF-8"
LC_MESSAGES="fr_CH.UTF-8"
LC_PAPER="fr_CH.UTF-8"
LC_NAME="fr_CH.UTF-8"
LC_ADDRESS="fr_CH.UTF-8"
LC_TELEPHONE="fr_CH.UTF-8"
LC_MEASUREMENT="fr_CH.UTF-8"
LC_IDENTIFICATION="fr_CH.UTF-8"
LC_ALL=

**Additional context**

contents of `$HOME/.config/bashtop/error.log` :

 
New instance of bashtop version: 0.9.9 Pid: 9598
 
New instance of bashtop version: 0.9.9 Pid: 7284
 
New instance of bashtop version: 0.9.9 Pid: 20463
 
New instance of bashtop version: 0.9.9 Pid: 17191
 
New instance of bashtop version: 0.9.10 Pid: 5405
 
New instance of bashtop version: 0.9.10 Pid: 21735
 
New instance of bashtop version: 0.9.10 Pid: 9154
21:37:31 ERROR: On line 2681 
 
New instance of bashtop version: 0.9.10 Pid: 28051
21:51:09 ERROR: On line 2889 
21:51:09 ERROR: On line 2889 
 
New instance of bashtop version: 0.9.10 Pid: 28722
 
New instance of bashtop version: 0.9.12 Pid: 18600
 
New instance of bashtop version: 0.9.12 Pid: 5943
/usr/bin/bashtop: line 1505: -: syntax error: operand expected (error token is "-")
 
New instance of bashtop version: 0.9.12 Pid: 7015
/usr/bin/bashtop: line 1505: -: syntax error: operand expected (error token is "-")
 
New instance of bashtop version: 0.9.12 Pid: 7327
/usr/bin/bashtop: line 1505: -: syntax error: operand expected (error token is "-")
 
New instance of bashtop version: 0.9.12 Pid: 7810
/usr/bin/bashtop: line 1505: -: syntax error: operand expected (error token is "-")
 
New instance of bashtop version: 0.9.12 Pid: 11436
 
New instance of bashtop version: 0.9.12 Pid: 11913
/usr/bin/bashtop: line 1505: -: syntax error: operand expected (error token is "-")
 
New instance of bashtop version: 0.9.12 Pid: 20986
/usr/bin/bashtop: line 1505: -: syntax error: operand expected (error token is "-")
 
New instance of bashtop version: 0.9.12 Pid: 21714
/usr/bin/bashtop: line 1505: -: syntax error: operand expected (error token is "-")
 
New instance of bashtop version: 0.9.12 Pid: 22378
/usr/bin/bashtop: line 1505: -: syntax error: operand expected (error token is "-")
 
New instance of bashtop version: 0.9.12 Pid: 22790
/usr/bin/bashtop: line 1505: -: syntax error: operand expected (error token is "-")
 
New instance of bashtop version: 0.9.12 Pid: 23184
/usr/bin/bashtop: line 1505: -: syntax error: operand expected (error token is "-")
 
New instance of bashtop version: 0.9.12 Pid: 23362
/usr/bin/bashtop: line 1505: -: syntax error: operand expected (error token is "-")
 
New instance of bashtop version: 0.9.12 Pid: 23721
/usr/bin/bashtop: line 1505: -: syntax error: operand expected (error token is "-")
 
New instance of bashtop version: 0.9.12 Pid: 24003
/usr/bin/bashtop: line 1505: -: syntax error: operand expected (error token is "-")
 
New instance of bashtop version: 0.9.12 Pid: 26085
 
New instance of bashtop version: 0.9.12 Pid: 26927
/usr/bin/bashtop: line 1505: -: syntax error: operand expected (error token is "-")
 
New instance of bashtop version: 0.9.12 Pid: 28862
/usr/bin/bashtop: line 1505: -: syntax error: operand expected (error token is "-")

**Comment by Phil995511** [comment]
After testing the bashtop_0.9.12_all.deb version from http://packages.azlux.fr/debian/dists/buster/ crash her too ;-(

The 0.9.11 version from Debian repertory seems more stable https://packages.debian.org/sid/all/bashtop/download

**Comment by aristocratos** [comment]
I've added some extended error checking in v0.9.13
Try updating, hopefully we will get some useful output in the error.log if the problem persists.

**Comment by Phil995511** [comment]
Ok I will give you more details if the problem persists. I must to use ' bashtop -v ' for start the  extended error checking ?? The 0.9.13 update is not more disponible on Ubuntu PPA, the lasted version 0.9.12+git229-f48b59e~ununtu20.10.1 are updated from this night but it is bugged. The 0.9.11 version from Debian SID are really very stable. Regards.

**Comment by aristocratos** [comment]
If you want the latest updates faster install manually from git, I don't manage the distribution packages.
And you only have to run `bashtop` the regular way, the extended error checking is on as long as error logging is activated, which it is by default.

The errors is very likely linked to the update to psutil in version 0.9.12, in version 0.9.3 bashtop will fallback to using /proc instead of psutil if there are errors and you are on linux.

But even if it works on the new version, it would be really helpful if you updated this issue with your error.log again, so I can fix any underlying issues with the psutil script.

**Comment by Phil995511** [comment]
The 0.9.12 version (from Ubuntu ppa) works wonderfully under Raspbian OS ARM 64 Bits Beta (RPi4). On Debian 10 Buster I have added sid repository availability for this software, I'm happy with the version 0.9.11, it work fine. When I have a moment I will test 0.9.13 on my laptop. Regards

**Comment by Phil995511** [comment]
I have imported the code from :

https://github.com/aristocratos/bashtop/blob/master/bashtop

Your new 0.9.13 version run perfectly on Ubuntu 20.04 :)

I hope maintainers will update their directories quickly...

Good week-end and best regards to you.

**Comment by Phil995511** [comment]
**

> Additional context

**

contents of $HOME/.config/bashtop/error.log :

New instance of bashtop version: 0.9.13 Pid: 13359
/usr/lib/python3/dist-packages/psutil/_pslinux.py:1222: RuntimeWarning: ignoring FileNotFoundError(2, 'No such file or directory') for file '/sys/class/hwmon/hwmon0/temp1_input'
  warnings.warn("ignoring %r for file %r" % (err, path),
/usr/lib/python3/dist-packages/psutil/_pslinux.py:1222: RuntimeWarning: ignoring FileNotFoundError(2, 'No such file or directory') for file '/sys/class/hwmon/hwmon0/temp2_input'
  warnings.warn("ignoring %r for file %r" % (err, path),
/usr/lib/python3/dist-packages/psutil/_pslinux.py:1222: RuntimeWarning: ignoring FileNotFoundError(2, 'No such file or directory') for file '/sys/class/hwmon/hwmon2/temp1_input'
  warnings.warn("ignoring %r for file %r" % (err, path),
/usr/lib/python3/dist-packages/psutil/_pslinux.py:1222: RuntimeWarning: ignoring FileNotFoundError(2, 'No such file or directory') for file '/sys/class/hwmon/hwmon2/temp2_input'
  warnings.warn("ignoring %r for file %r" % (err, path),
/usr/lib/python3/dist-packages/psutil/_pslinux.py:1222: RuntimeWarning: ignoring OSError(61, 'No data available') for file '/sys/class/hwmon/hwmon5/temp1_input'
  warnings.warn("ignoring %r for file %r" % (err, path),
/home/philippe/Bureau/bashtop: line 1505: -: syntax error: operand expected (error token is "-")
 
New instance of bashtop version: 0.9.13 Pid: 13671
/usr/lib/python3/dist-packages/psutil/_pslinux.py:1222: RuntimeWarning: ignoring FileNotFoundError(2, 'No such file or directory') for file '/sys/class/hwmon/hwmon0/temp1_input'
  warnings.warn("ignoring %r for file %r" % (err, path),
/usr/lib/python3/dist-packages/psutil/_pslinux.py:1222: RuntimeWarning: ignoring FileNotFoundError(2, 'No such file or directory') for file '/sys/class/hwmon/hwmon0/temp2_input'
  warnings.warn("ignoring %r for file %r" % (err, path),
/usr/lib/python3/dist-packages/psutil/_pslinux.py:1222: RuntimeWarning: ignoring FileNotFoundError(2, 'No such file or directory') for file '/sys/class/hwmon/hwmon2/temp1_input'
  warnings.warn("ignoring %r for file %r" % (err, path),
/usr/lib/python3/dist-packages/psutil/_pslinux.py:1222: RuntimeWarning: ignoring FileNotFoundError(2, 'No such file or directory') for file '/sys/class/hwmon/hwmon2/temp2_input'
  warnings.warn("ignoring %r for file %r" % (err, path),
/usr/lib/python3/dist-packages/psutil/_pslinux.py:1222: RuntimeWarning: ignoring OSError(61, 'No data available') for file '/sys/class/hwmon/hwmon5/temp1_input'
  warnings.warn("ignoring %r for file %r" % (err, path),
/home/philippe/Bureau/bashtop: line 1505: -: syntax error: operand expected (error token is "-")
 
New instance of bashtop version: 0.9.13 Pid: 13876
/usr/lib/python3/dist-packages/psutil/_pslinux.py:1222: RuntimeWarning: ignoring FileNotFoundError(2, 'No such file or directory') for file '/sys/class/hwmon/hwmon0/temp1_input'
  warnings.warn("ignoring %r for file %r" % (err, path),
/usr/lib/python3/dist-packages/psutil/_pslinux.py:1222: RuntimeWarning: ignoring FileNotFoundError(2, 'No such file or directory') for file '/sys/class/hwmon/hwmon0/temp2_input'
  warnings.warn("ignoring %r for file %r" % (err, path),
/usr/lib/python3/dist-packages/psutil/_pslinux.py:1222: RuntimeWarning: ignoring FileNotFoundError(2, 'No such file or directory') for file '/sys/class/hwmon/hwmon2/temp1_input'
  warnings.warn("ignoring %r for file %r" % (err, path),
/usr/lib/python3/dist-packages/psutil/_pslinux.py:1222: RuntimeWarning: ignoring FileNotFoundError(2, 'No such file or directory') for file '/sys/class/hwmon/hwmon2/temp2_input'
  warnings.warn("ignoring %r for file %r" % (err, path),
/usr/lib/python3/dist-packages/psutil/_pslinux.py:1222: RuntimeWarning: ignoring OSError(61, 'No data available') for file '/sys/class/hwmon/hwmon5/temp1_input'
  warnings.warn("ignoring %r for file %r" % (err, path),
/home/philippe/Bureau/bashtop: line 1505: -: syntax error: operand expected (error token is "-")

**Comment by aristocratos** [comment]
@Phil995511 
Possible fix and additional error checking added in v0.9.14

**Comment by Phil995511** [comment]
Hello, after 3 hours of running the 0.9.14 version all is ok, I dont have of crash :) 

The CPU consumption has greatly decreased with this new version, congratulations on that !! It will soon be as light as 'htop'.

It's possible to close this ticket :)

Best regards and good continuation for you !!!
